CRM/Core/BAO/MessageTemplate.php

```diff
@@ -406,6 +406,9 @@ public static function sendTemplate($params) {
       'text' => $dao->text,
       'html' => $dao->html,
       'format' => $dao->format,
+      'groupName' => $params['groupName'],
+      'valueName' => $params['valueName'],
+      'messageTemplateID' => $params['messageTemplateID'],
```

Comment:
At this point the $params will always have these 3 information as there is a check at L365 above.